src/app/view/perspectives/kanban/pipes/kanban-column-titles.pipe.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Lumeer: Modern Data Definition and Processing Platform
+ *
+ * Copyright (C) since 2017 Lumeer.io, s.r.o. and/or its affiliates.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import {Pipe, PipeTransform} from '@angular/core';
+import {KanbanAttribute, KanbanConfig} from '../../../../core/store/kanbans/kanban';
+
+@Pipe({
+  name: 'kanbanColumnTitles',
+})
+export class KanbanColumnTitlesPipe implements PipeTransform {
+  public transform(config: KanbanConfig, attribute: KanbanAttribute): string[] {
+    const res = ((config && config.columns) || [])
+      .filter(column => this.containsAttribute(column.createdFromAttributes, attribute))
+      .map(c => c.title);
+
+    return res;
+  }
+
+  private containsAttribute(attributes: KanbanAttribute[], attribute: KanbanAttribute): boolean {
+    return attributes.map(a => a.attributeId).indexOf(attribute.attributeId) >= 0;
```

Comment:
Always use `(array || [])` construction in situations like this, because it's not guaranteed, that attributes are defined.